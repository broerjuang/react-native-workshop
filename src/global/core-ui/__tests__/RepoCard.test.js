//@flow
import React from 'react';
import renderer from 'react-test-renderer';
import {RepoCard} from '../index';
import type {Repo} from '../RepoCard';
describe('RepoCard', () => {
  it('should render RepoCard for repo corectly', () => {
    let repo: Repo = {
      full_name: 'astridtamara/bootcamp',
      description: 'KodeFox Bootcamp',
      stargazers_count: 0,
      forks_count: 1,
      language: 'JavaScript',
      fork: false,
    };

    let component = renderer.create(
      <RepoCard repo={repo} onPress={() => {}} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render RepoCard for forked repo corectly', () => {
    let repo = {
      full_name: 'astridtamara/kfbootcamp',
      description: '',
      stargazers_count: 0,
      forks_count: 0,
      language: 'JavaScript',
      fork: true,
    };

    let component = renderer.create(
      <RepoCard repo={repo} onPress={() => {}} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render RepoCard for null langColor', () => {
    let repo = {
      full_name: 'astridtamara/kfbootcamp',
      description: '',
      stargazers_count: 0,
      forks_count: 0,
      language: 'Alpine Abuild',
      fork: true,
    };

    let component = renderer.create(
      <RepoCard repo={repo} onPress={() => {}} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render RepoCard for unknown language', () => {
    let repo = {
      full_name: 'astridtamara/kfbootcamp',
      description: '',
      stargazers_count: 0,
      forks_count: 0,
      language: '----',
      fork: true,
    };

    let component = renderer.create(
      <RepoCard repo={repo} onPress={() => {}} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
