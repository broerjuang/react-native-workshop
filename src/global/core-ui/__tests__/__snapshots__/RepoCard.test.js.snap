// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoCard should render RepoCard for forked repo corectly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#d6d4d4",
      "borderTopWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "padding": 12,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "width": "85%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginTop": 4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginRight": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 12,
              "marginLeft": 6,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginRight": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 12,
              "marginLeft": 6,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginRight": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 12,
              "marginLeft": 6,
            }
          }
        >
          JavaScript
        </Text>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
</View>
`;

exports[`RepoCard should render RepoCard for repo corectly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#d6d4d4",
      "borderTopWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "padding": 12,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "width": "85%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      KodeFox Bootcamp
    </Text>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginTop": 4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginRight": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 12,
              "marginLeft": 6,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginRight": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 12,
              "marginLeft": 6,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginRight": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 12,
              "marginLeft": 6,
            }
          }
        >
          JavaScript
        </Text>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
</View>
`;
